inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java

```diff
@@ -191,28 +175,29 @@ private void addMetric(boolean result, long size, Event event) {
 
     /**
      * responsePackage
-     * 
+     *
+     * @param ctx
      * @param  code
-     * @param  e
      * @throws Exception
      */
-    private void responsePackage(ResultCode code, MessageEvent e)
+    private void responsePackage(ChannelHandlerContext ctx, ResultCode code)
             throws Exception {
         ResponseInfo.Builder builder = ResponseInfo.newBuilder();
         builder.setResult(code);
 
         // encode
         byte[] responseBytes = builder.build().toByteArray();
         //
-        ChannelBuffer buffer = ChannelBuffers.wrappedBuffer(responseBytes);
-
-        Channel remoteChannel = e.getChannel();
+        ByteBuf buffer = ByteBufAllocator.DEFAULT.buffer(responseBytes.length);
+        buffer.writeBytes(responseBytes);
```

Comment:
It is better to use Unpooled.wrappedBuffer(byte[]).

Reply:
done